builder/module-docs.nix

```diff
@@ -0,0 +1,34 @@
+{ pkgs, lib, ... }:
+let
+  # Inspired by https://github.com/astro/microvm.nix/blob/e8d5f12b834a59187c7ec147a8952a0567f97939/pkgs/doc.nix
+  extraConfig = { lib, ... }: {
+    options._module.args = lib.mkOption {
+      internal = true; # Hide `_module` from the docs
+    };
+
+    config._module = {
+      check = false; # Don't check for unset variables
+      args = {
+        inherit pkgs lib; # Provide pkgs and lib to modules
+      };
+    };
+  };
+
+  optionsFor = module:
+    (lib.evalModules {
+      modules = [
+        extraConfig
+        module
+      ];
+    }).options;
+
+  makeOptionsDoc = module: (pkgs.nixosOptionsDoc {
+    options = optionsFor module;
+    transformOptions = options: builtins.removeAttrs options [ "declarations" ];
```

Comment:
We probably want to filter declarations separately and also provide the optionsJSON to easily integrate it into nüschtos search.

Reply:
I'm not familiar with "nüschtos search". Do you have an example of how the module should look like?

Reply:
It should look similar to this https://github.com/NuschtOS/search/blob/3051be7f403bff1d1d380e4612f0c70675b44fc9/nix/wrapper.nix#L4-L17